pkg/apiserver/authn.go

```diff
@@ -28,11 +28,12 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/auth/authenticator/request/basicauth"
 	"github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/auth/authenticator/request/union"
 	"github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/auth/authenticator/request/x509"
+	"github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/auth/authenticator/token/openid"
 	"github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/auth/authenticator/token/tokenfile"
 )
 
 // NewAuthenticator returns an authenticator.Request or an error
-func NewAuthenticator(basicAuthFile, clientCAFile, tokenFile, serviceAccountKeyFile string, serviceAccountLookup bool, helper tools.EtcdHelper) (authenticator.Request, error) {
+func NewAuthenticator(basicAuthFile, clientCAFile, tokenFile, openIDClientID, openIDEndpoint, serviceAccountKeyFile string, serviceAccountLookup bool, helper tools.EtcdHelper) (authenticator.Request, error) {
```

Comment:
I probably should have done it when I added the service account options, but this is probably at the point where we need an Options struct with sub-structs for each authenticator type.

Basic auth: basicAuthFile
Cert auth: clientCAFile
CSV token: tokenFile
OpenID token: openIDClientID, openIdEndpoint
Service account token: serviceAccountKeyFile, serviceAccountLookup, helper